client/consensus/epochs/src/lib.rs

```diff
@@ -518,13 +518,13 @@ where
 		let is_descendent_of = descendent_of_builder.build_is_descendent_of(None);
 
 		let predicate = |epoch: &PersistedEpochHeader<E>| match *epoch {
-			PersistedEpochHeader::Genesis(ref epoch_0, _) => epoch_0.start_slot <= slot,
+			PersistedEpochHeader::Genesis(_, ref epoch_1) => epoch_1.start_slot <= slot,
```

Comment:
Why do we compare with start slot and not end slot?

Reply:
Okay, I found out why. The predicate is used to identify the current epoch of the finalized chain.